test/tests/preconditioners/pbp/pbp_test.i

```diff
@@ -31,6 +29,11 @@
   [../]
 []
 
+[Problem]
+  type = FEProblem
+  error_on_jacobian_nonzero_reallocation = true
```

Comment:
Could you add this to the DG version as well? I'm more worried about that one...

Reply:
I was intending to add this to DG version. I looked at the wrong file :)